maltoolbox/attackgraph/attackgraph.py

```diff
@@ -165,7 +165,9 @@ class AttackGraph():
     """Graph representation of attack steps"""
     def __init__(self, lang_graph = None, model: Optional[Model] = None):
         self.nodes = []
+        self.id_to_node = {}  # to get nodes by id faster
```

Comment:
I think there is a bigger issue in the current implementation where lists are preferred over dicts. This forces the code to search in lists if e.g. a specific assets needs to be found. It would make the code both simpler and faster if dicts where used that were based by the asset id or name, association name, etc. Maybe it is worth rewriting the toolbox around dicts?

Reply:
Yes, maybe we can replace the list all together even!
Also think this should be a new issue/PR because it might be many changes to the code unrelated to optimizations.

Reply:
A potential issue with deferring that I see is that this task may change the API of the toolbox and introduce breaking changes. But you know what priorities are there.

Reply:
Could be an issue if someone accesses .assets or .nodes, that is true..